include/roboptim/core/detail/structured-input.hxx

```diff
@@ -0,0 +1,81 @@
+// Copyright (C) 2015 by Grégoire Duchemin, AIST, CNRS, EPITA
+//
+// This file is part of the roboptim.
+//
+// roboptim is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// roboptim is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with roboptim.  If not, see <http://www.gnu.org/licenses/>.
+
+#ifndef ROBOPTIM_CORE_DETAIL_STRUCTURED_INPUT_HXX
+# define ROBOPTIM_CORE_DETAIL_STRUCTURED_INPUT_HXX
+
+template<typename Derived>
+Eigen::Block<Derived>
+getBlock (Eigen::SparseMatrixBase<Derived>& m, int colStart, int colLength)
+{
+  return Eigen::Block<Derived> (m.derived (), 0, colStart, m.rows (), colLength);
+}
```

Comment:
I guess a comment explaining why you needed that would be good (since that may be fixed in Eigen later on).